common/src/main/java/com/microsoft/identity/common/internal/broker/InactiveBrokerClient.java

```diff
@@ -0,0 +1,166 @@
+    private String mCallingPackageName;
+    private BrokerAccountServiceConnection mBrokerAccountServiceConnection;
+    private Intent mBrokerAccountServiceIntent;
+    private Boolean mBound = false;
+
+    // Must be called from BG thread.
+    public static String getSerializedKeyFromInactiveBroker(@NonNull Context context,
+                                                            @NonNull final String callingPackageName) {
+        final String methodName = ":getKeyFromInactiveBroker";
+
+        final InactiveBrokerClient client = new InactiveBrokerClient(context, callingPackageName);
+        try {
+            final BrokerAccountServiceFuture brokerAccountServiceFuture = client.connect();
+            IBrokerAccountService service = brokerAccountServiceFuture.get();
+
+            Bundle resultBundle = service.getInactiveBrokerKey();
+
+            // This means that the inactive broker doesn't support getInactiveBrokerKey().
+            if (resultBundle == null){
+                Logger.verbose(TAG + methodName, "resultBundle is null");
+                return null;
+            }
+
+            return resultBundle.getString(AuthenticationConstants.Broker.BROKER_KEYSTORE_SYMMETRIC_KEY);
+
+        } catch (final BaseException | InterruptedException | ExecutionException | RemoteException e) {
+            Logger.error(
+                    TAG + methodName,
+                    "Exception is thrown when trying to get key from inactive broker:"
+                            + e.getMessage(),
+                    e);
+        } finally {
+            client.disconnect();
+        }
+
+        return null;
+    }
+
+    /**
+     * Constructor for the BrokerAccountServiceClient
+     *
+     * @param context
+     * @param callingPackageName name of the calling package.
+     */
+    private InactiveBrokerClient(@NonNull final Context context,
+                                 @NonNull final String callingPackageName) {
+        mContext = context;
+        mCallingPackageName = callingPackageName;
+        mBrokerAccountServiceIntent = getIntentForBrokerAccountService(mContext);
+    }
+
+    /**
+     * Binds to the service and returns a future that provides the proxy for the calling the BrokerAccountService
+     *
+     * @return MicrosoftAuthServiceFuture
+     */
+    private BrokerAccountServiceFuture connect() throws ClientException {
+
+        if (mBrokerAccountServiceIntent == null) {
+            throw new ClientException("mBrokerAccountServiceIntent is null. BrokerAccountService.");
+        }
+
+        BrokerAccountServiceFuture future = new BrokerAccountServiceFuture();
+        mBrokerAccountServiceConnection = new BrokerAccountServiceConnection(future);
+
+        mBound = mContext.bindService(mBrokerAccountServiceIntent, mBrokerAccountServiceConnection, Context.BIND_AUTO_CREATE);
+        Logger.verbose(TAG + "connect", "The status for MicrosoftAuthService bindService call is: " + Boolean.valueOf(mBound));
```

Comment:
>MicrosoftAuthService [](start = 56, length = 20)

BrokerAccountService?

Reply:
@rpdome unresolving this as this is still not fixed after the `Resolve comments` commit